### PR TITLE
Migrate tests to use vitest

### PR DESCRIPTION
**Pull request by hwbllmnn**
## Description

See title.

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

@geostyler/devs Please review.

### CODE REVIEW

File: jest/setup.js
Comment (KaiVolland) [inline jest/setup.js:1]: Maybe the folder should be renamed to `test/setup.js` or similiar.

File: vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    server: {
+      deps: {
+        inline: true
+      }
+    },
+    setupFiles: ['./jest/setup.js'],
```
Comment (KaiVolland) [inline vitest.config.ts:10]:
```suggestion
    setupFiles: ['./test/setup.js'],
```

Comment (hwbllmnn) [reply vitest.config.ts]: done

**Review by KaiVolland — APPROVED** [review]

**Pull request merged**